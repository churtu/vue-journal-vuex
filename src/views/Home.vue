<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
    <button @click="goToDaybook" class="btn-primary btn">Daybook</button>
  <button class="btn-secondary btn">action</button>
  <button class="btn-info btn">action</button>
  <button class="btn-warning btn">action</button>
  <button class="btn-danger btn">action</button>
  <button class="btn-dark btn">action</button>
  </div>
</template> 
<script>
export default {
  methods:{
    goToDaybook() {
    this.$router.push({name:'no-entry'})
  },
  }
}
</script>