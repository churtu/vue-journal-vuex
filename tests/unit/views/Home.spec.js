import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home'


describe('Home view', () => {

    test('Debe renderizar el componente correctamente', () => {
        const wrapper = shallowMount(Home)
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Hacer click en un boton debe renderizar a no-entry', () => {
        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount(Home, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })
        
        wrapper.find('button').trigger('click')
        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({name: 'no-entry'})
    })  
    
})