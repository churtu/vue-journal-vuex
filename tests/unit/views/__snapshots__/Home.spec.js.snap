// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view Debe renderizar el componente correctamente 1`] = `<div class="home"><img alt="Vue logo" src=""><button class="btn-primary btn">Daybook</button><button class="btn-secondary btn">action</button><button class="btn-info btn">action</button><button class="btn-warning btn">action</button><button class="btn-danger btn">action</button><button class="btn-dark btn">action</button></div>`;
